### PR TITLE
Feature/271-delete-redundant-files

### PR DESCRIPTION
**Pull request by waltersajtos**
## Description

Currently, there will be files in the filesystem and database that aren't linked to any project. The changes I made should fix this.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [ ] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [ ] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Create a project with a project icon
2. Delete/ update the project with another icon
3. The old icon should be deleted from the database

#271 

Closes: #271

### CODE REVIEW

File: API/Controllers/ProjectController.cs [hunk 1]
```diff
@@ -56,7 +57,15 @@ public class ProjectController : ControllerBase
...
         /// <param name="mapper">The mapper which is used to convert the resources to the models to the resource results.</param>
-        public ProjectController(IProjectService projectService, IUserService userService, IFileService fileService, IMapper mapper, IFileUploader fileUploader)
+        /// <param name="authorizationHelper">The authorization helper which is used to communicate with the authorization helper class.</param>
+        /// <param name="userProjectService">The user project service is responsible for users that are following / liking projects.</param>
+        public ProjectController(IProjectService projectService,
+                                 IUserService userService,
+                                 IMapper mapper,
+                                 IAuthorizationHelper authorizationHelper,
+                                 IFileService fileService,
+                                 IFileUploader fileUploader,
```
Comment (Brend-Smits) [inline API/Controllers/ProjectController.cs:67]: "Parameter 'fileUploader' has no matching param tag in the XML comment for 'ProjectController.ProjectController(IProjectService, IUserService, IMapper, IAuthorizationHelper, IFileService, IFileUploader, IUserProjectService)' (but other parameters do)
"

File: API/Controllers/ProjectController.cs [hunk 2]
```diff
@@ -252,41 +261,68 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
...
+                        Detail = "The specified file was not found while updating project.",
+                        Instance = "69166D3D-6D34-4050-BD25-71F1BEBE43D3"
+                    };
+                    return BadRequest(problem);
+                }  
             }
 
-            // Check if the icon has been changed
-            if(projectResource.FileId != project.ProjectIconId)
+            /*// Check if the icon has been changed
```
Comment (Brend-Smits) [inline API/Controllers/ProjectController.cs:313]: Commented out code?

File: API/Controllers/ProjectController.cs [hunk 3]
```diff
@@ -252,41 +261,68 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
...
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+            bool isAllowed = userService.UserHasScope(user.IdentityId, nameof(Defaults.Scopes.ProjectWrite));
+
+            if(!(project.UserId == user.Id || isAllowed))
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to edit the project.",
+                    Detail = "The user is not allowed to edit the project.",
+                    Instance = "2E765D18-8EBC-4117-8F9E-B800E8967038"
```
Comment (Brend-Smits) [inline API/Controllers/ProjectController.cs]: GUID is not unique, generate a fresh one using Visual Studio or an online tool please.

File: API/Controllers/ProjectController.cs [hunk 4]
```diff
@@ -252,41 +261,68 @@ public async Task<IActionResult> UpdateProject(int projectId, [FromBody] Project
...
+                {
+                    project.ProjectIcon = file;
+
+                } else
+                {
+                    ProblemDetails problem = new ProblemDetails
+                    {
+                        Title = "File was not found.",
+                        Detail = "The specified file was not found while updating project.",
+                        Instance = "69166D3D-6D34-4050-BD25-71F1BEBE43D3"
```
Comment (Brend-Smits) [inline API/Controllers/ProjectController.cs:307]: GUID is not unique, generate a fresh one using Visual Studio or an online tool, please.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]

**Review by niraymak — CHANGES_REQUESTED** [review]
Tests work! Except for Brend's request no more requests for me. 👍

**Review by Brend-Smits — APPROVED** [review]
Awesome! Thank you :)

**Review by niraymak — APPROVED** [review]
Good work 👍

**Review by Brend-Smits — APPROVED** [review]

**Review by niraymak — APPROVED** [review]

**Pull request merged**